src/modules/LoginScreen.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useState } from 'react';
+import { Input } from '../components/Input';
+
+export const LoginScreen: React.FC = () => {
+  const [fields, setFields] = useState({ email: '', password: '' });
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFields({
```

Comment:
Aqui dois pontos:
Nesse caso aqui não tem problema dois elementos alterarem um state, mas dependendo do caso, por exemplo com chamadas de API, isso pode causar um rerender infinto.
Sempre que for utilizar o estado anterior para realizar uma atualização de estado, o melhor é fazer uma função ao invés de usar direto o estado anterior.
https://reactjs.org/docs/hooks-reference.html#functional-updates

Reply:
Sempre reforçam isso nas explicações de setState, mas me distraí e não coloquei 🤡 . Vou arrumar